File: backend/app/models/vendor.py
from sqlalchemy import Column, Integer, String, ForeignKey, Enum, Text, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base
import enum

class VerificationStatus(str, enum.Enum):
    PENDING = "PENDING"
    VERIFIED = "VERIFIED"
    REJECTED = "REJECTED"

class Vendor(Base):
    __tablename__ = "vendors"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), unique=True, nullable=False)
    phone_number = Column(String, unique=True, index=True)
    company_name = Column(String, index=True)
    office_address = Column(Text)
    verification_status = Column(Enum(VerificationStatus), default=VerificationStatus.PENDING, index=True)
    vendor_uid = Column(String, unique=True, index=True, nullable=True)
    
    # Personal details
    dob = Column(String, nullable=True)
    gender = Column(String, nullable=True)
    
    # Company details
    website = Column(String, nullable=True)
    gstin = Column(String, nullable=True)
    description = Column(Text, nullable=True)
    logo_url = Column(String, nullable=True)
    
    # QR Code fields (added via migration)
    # Note: If migration hasn't run, these will cause errors
    # Run: python check_and_fix_qr_columns.py or alembic upgrade head
    qr_code_data = Column(Text, nullable=True)  # JSON string with company details
    qr_code_image_url = Column(String, nullable=True)  # URL to QR code image
    qr_code_generated_at = Column(DateTime(timezone=True), nullable=True)
    
    user = relationship("User", backref="vendor_profile")
    documents = relationship("Document", back_populates="vendor")
    visits = relationship("Visit", back_populates="vendor")

File: vms/lib/core/models/vendor.dart
class Vendor {
  final int id;
  final int userId;
  final String? phoneNumber;
  final String? companyName;
  final String? officeAddress;
  final String verificationStatus;
  final String? vendorUid;
  final String? dob;
  final String? gender;
  final String? website;
  final String? gstin;
  final String? description;
  final String? logoUrl;
  final String? email;
  final String? fullName;

  Vendor({
    required this.id,
    required this.userId,
    this.phoneNumber,
    this.companyName,
    this.officeAddress,
    required this.verificationStatus,
    this.vendorUid,
    this.dob,
    this.gender,
    this.website,
    this.gstin,
    this.description,
    this.logoUrl,
    this.email,
    this.fullName,
  });

  factory Vendor.fromJson(Map<String, dynamic> json) {
    // Handle verification_status - ensure it's a string
    String verificationStatus = 'PENDING';
    if (json['verification_status'] != null) {
      final status = json['verification_status'];
      if (status is String) {
        verificationStatus = status;
      } else {
        // If it's an enum or object, convert to string
        verificationStatus = status.toString().toUpperCase();
        // Remove enum prefix if present (e.g., "VerificationStatus.VERIFIED" -> "VERIFIED")
        if (verificationStatus.contains('.')) {
          verificationStatus = verificationStatus.split('.').last;
        }
      }
    }
    
    return Vendor(
      id: json['id'],
      userId: json['user_id'],
      phoneNumber: json['phone_number'],
      companyName: json['company_name'],
      officeAddress: json['office_address'],
      verificationStatus: verificationStatus,
      vendorUid: json['vendor_uid'],
      dob: json['dob'],
      gender: json['gender'],
      website: json['website'],
      gstin: json['gstin'],
      description: json['description'],
      logoUrl: json['logo_url'],
      email: json['email'],
      fullName: json['full_name'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'phone_number': phoneNumber,
      'company_name': companyName,
      'office_address': officeAddress,
      'verification_status': verificationStatus,
      'vendor_uid': vendorUid,
      'dob': dob,
      'gender': gender,
      'website': website,
      'gstin': gstin,
      'description': description,
      'logo_url': logoUrl,
      'email': email,
      'full_name': fullName,
    };
  }
}
